src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -167,6 +186,64 @@ public ClientListPanel getClientListPanel() {
         return clientListPanel;
     }
 
+    /**
+     * Applies the appropriate theme (light or dark) to the scene based on the current theme setting.
+     *
+     * If the theme is "light", the LIGHT_THEME stylesheet is applied.
+     * Otherwise, the DARK_THEME stylesheet is applied.
+     */
+    public void applyTheme() {
+        scene.getStylesheets().clear(); // Clear any previously applied styles
+
+        if (theme.equalsIgnoreCase(LIGHT_THEME)) {
+            scene.getStylesheets().add(getClass().getResource(LIGHT_THEME_PATH).toExternalForm());
+        } else {
+            scene.getStylesheets().add(getClass().getResource(DARK_THEME_PATH).toExternalForm());
+        }
+    }
+
+    /**
+     * Handles switching the theme from dark to light mode.
+     * If the theme is already set to light mode, it simply notifies the user without any changes.
+     */
+    @FXML
+    private void handleLightTheme() {
+        if (theme.equals(DARK_THEME)) {
+            scene.getStylesheets().remove(getClass().getResource(DARK_THEME_PATH).toExternalForm());
+            scene.getStylesheets().add(getClass().getResource(LIGHT_THEME_PATH).toExternalForm());
+            theme = LIGHT_THEME;
+
+            // Get user prefs, set the theme, and update
+            UserPrefs updatedPrefs = (UserPrefs) logic.getUserPrefs();
+            logic.setUserPrefs(updatedPrefs.setTheme(LIGHT_THEME));
+
+            resultDisplay.setFeedbackToUser("Theme changed to " + theme + " mode.");
+        } else {
+            resultDisplay.setFeedbackToUser("Theme is already set to " + theme + " mode.");
+        }
+    }
+
+    /**
+     * Handles switching the theme from light to dark mode.
+     * If the theme is already set to dark, it simply notifies the user without any changes.
+     */
+    @FXML
+    private void handleDarkTheme() {
+        if (theme.equals(LIGHT_THEME)) {
+            scene.getStylesheets().remove(getClass().getResource(LIGHT_THEME_PATH).toExternalForm());
+            scene.getStylesheets().add(getClass().getResource(DARK_THEME_PATH).toExternalForm());
+            theme = DARK_THEME;
+
+            // Get user prefs, set the theme, and update
+            UserPrefs updatedPrefs = (UserPrefs) logic.getUserPrefs();
+            logic.setUserPrefs(updatedPrefs.setTheme(DARK_THEME));
+
+            resultDisplay.setFeedbackToUser("Theme changed to " + theme + " mode.");
+        } else {
+            resultDisplay.setFeedbackToUser("Theme is already set to " + theme + " mode.");
```

Comment:
Is there a way to either abstract out the similar code, or handling this as a toggle? It seems like a lot of lines are being repeated, which may make maintenance difficult later. What do you think?

Reply:
I think having `public static final String THEME_CHANGE_MESSAGE = "Theme changed to %1$s mode.` will be a good way to abstract it out, as well as keeping our String formatting in line with our other messages

Reply:
Great suggestion, will make use of that!